### PR TITLE
feat: footer, light mode, and component consistency 

### PR DESCRIPTION
**Pull request by quinchs**
## Summary
This PR adds the footer and changes the component/typography tree to be consistent

### CODE REVIEW

**Comment by quinchs** [comment]
Light mode added and colors consistency

**Review by MeguminSama — COMMENTED** [review]
The sidebar icons should really be the same/similar colour as the discord logo and theme toggle button. They have better contrast, and has more consistency.

image

File: tsconfig.json [hunk 1]
```diff
@@ -13,7 +13,8 @@
     "resolveJsonModule": true,
     "isolatedModules": true,
     "jsx": "preserve",
-    "incremental": true
+    "incremental": true,
+    "experimentalDecorators": true
```
Comment (lhjt) [inline tsconfig.json:17]:
```suggestion
    "incremental": true
```

Is there a reason for this change? I can build the website fine without this compiler option.

File: pages/_app.tsx [hunk 2]
```diff
@@ -60,35 +75,49 @@ function DiscordDocsApp({ Component, pageProps, router }: AppProps) {
...
-          <Component {...pageProps} />
-        </ContentWrapper>
-      </Wrapper>
+      <ThemeProvider
+        disableTransitionOnChange
+        attribute="class"
+        value={{ light: lightTheme.className, dark: "dark-theme" }}
+        defaultTheme="system"
+      >
+        <NextNprogress color="#5865f2" />
```
Comment (lhjt) [inline pages/_app.tsx:84]: Are we sure we want to use the primary brand colour for the loader? It is virtually invisible on the summaries page when the user navigates to one of the summaries.

**Review by lhjt — CHANGES_REQUESTED** [review]
- The scroll track on the summaries page is inconsistent with the rest of the scroll tracks; it is rectangular at the top while the other ones are rounded:
image vs image

File: pages/_app.tsx (hunk 2 above)
Comment (quinchs) [reply pages/_app.tsx:84]: It should be a color within our theme

Comment (lhjt) [reply pages/_app.tsx:84]: So should we add a new colour to the theme? The fuchsia colour was selected directly from Discord's branding guidelines (and thus fits with the current blurple palette)

Comment (quinchs) [reply pages/_app.tsx:84]: Fair point, I mean im fine with it, going to cc @goverfl0w and @Liege72 for their opinion

**Review by MeguminSama — APPROVED** [review]

**Pull request merged**